src-common/vcastServer.ts

```diff
@@ -196,84 +196,55 @@ export async function transmitCommand(
+      transmitResponse.statusText = `Python interface error: \n   ${rawReturnData.data.text.join("\n   ")}`;
+    } else if (
+      rawReturnData.exitCode == pythonErrorCodes.couldNotStartClicastInstance
+    ) {
+      transmitResponse.success = false;
+      transmitResponse.statusText = `Server could not start clicast instance`;
+      setGLobalServerState(false);
+      await terminateServerProcessing(transmitResponse.statusText);
+    } else {
+      // Successful case
+      transmitResponse.success = true;
+      transmitResponse.statusText = `Enviro server response status: ${response.statusText}`;
+      transmitResponse.returnData = rawReturnData; // Return the raw data for further interpretation
+    }
+  } catch (error: any) {
+    if (
+      requestObject.command == vcastCommandType.shutdown ||
+      requestObject.command == vcastCommandType.ping
+    ) {
+      transmitResponse.success = false;
+    } else {
+      let errorText = "Unknown error";
+      if (error.message && error.message.includes("reason:")) {
+        errorText = error.message.split("reason:")[1].trim();
       } else {
-        transmitResponse.success = true;
-        transmitResponse.statusText = `Enviro server response status: ${response.statusText}`;
-        // the format of the data property is different based on the command
-        // so it is up to the caller to interpret it properly
-        transmitResponse.returnData = rawReturnData;
-        // there is always an exit code field but it is only used when
-        // executing tests or running clicast commands
+        errorText = `cannot communicate with server on port: ${SERVER_PORT}`;
       }
-    })
-    //
-    // if there is a communication error with the server it gets caught here
-    .catch(async (error) => {
-      // In the shutdown and ping cases we simply return a false success status,
-      // because in the shutdown case, the socket may be closed before  we
-      // read the response, and in the ping case, the server may not be
-      // ready yet and the caller will retry the ping.
-      if (
-        requestObject.command == vcastCommandType.shutdown ||
-        requestObject.command == vcastCommandType.ping
-      ) {
-        transmitResponse.success = false;
-      } else {
-        // For all other communication errors, we terminate server mode
-        let errorText = error.message.split("reason:")[1].trim();
-        // for some reason, when the server is down, the reason is blank
-        // so we insert a generic message in this case.
-        if (errorText.length === 0) {
-          errorText = `cannot communicate with server on port: ${SERVER_PORT}`;
-        }
-        let errorDetails = `command: ${requestObject.command}, error: ${errorText}`;
-        transmitResponse.success = false;
-        transmitResponse.statusText = `Enviro server error: ${errorDetails}`;
 
-        // In the core extension, this callback will shutdown the
```

Comment:
Can we check what's happened with these comments, and if we've lost "value" here by losing these comments?

Reply:
Added comments back again.